beacon-chain/sync/data_columns.go

```diff
@@ -48,6 +74,67 @@ func RequestDataColumnSidecarsByRoot(
 	p2p p2p.P2P,
 	ctxMap ContextByteVersions,
 	newColumnsVerifier verification.NewDataColumnsVerifier,
+) ([]blocks.VerifiedRODataColumn, error) {
+	log := log.WithFields(logrus.Fields{
+		"blockRoot":          fmt.Sprintf("%#x", block.Root()),
+		"dataColumnsToFetch": dataColumnsToFetch,
+	})
+	log.Debug("Attempting to fetch or reconstruct data columns")
+
+	// First, try to request the columns directly.
+	sidecars, err := OnlyRequestDataColumnSidecarsByRoot(
+		ctx,
+		dataColumnsToFetch,
+		block,
+		peers,
+		clock,
+		p2p,
+		ctxMap,
+		newColumnsVerifier,
+	)
+	if err == nil {
+		// Successfully fetched.
+		log.Debug("Successfully fetched requested data columns")
+		return sidecars, nil
+	}
+
+	// If the error is specifically UnavailableColumnsError, attempt reconstruction.
+	if errors.Is(err, &UnavailableColumnsError{}) {
+		log.WithError(err).Debug("Fetching failed due to no available peers, attempting reconstruction")
+		reconstructedSidecars, reconstructErr := ReconstructDataColumnsByRoot(
```

Comment:
Nit: Same, I guess here we can put all arguments on the same line.

Reply:
43a6e16954522054d5e030d4cd6cd77dbafc6493